contract/AElf.Contracts.Genesis/BasicContractZero_Helper.cs

```diff
@@ -362,13 +377,64 @@ private void AssertContractExists(Hash codeHash)
 
     private void AssertInlineDeployOrUpdateUserContract()
     {
-        Assert(Context.Origin == Context.Sender || !IsMainChain(), "Deploy or update contracts using inline transactions is not allowed.");
+        Assert(Context.Origin == Context.Sender || !IsMainChain(),
+            "Deploy or update contracts using inline transactions is not allowed.");
     }
 
     private bool IsMainChain()
     {
         return Context.GetContractAddressByName(SmartContractConstants.TreasuryContractSystemName) != null;
     }
+
+    private void CheckSignatureAndData(ContractOperation contractOperation, int version, Hash codeHash)
+    {
+        Assert(contractOperation.Deployer != null && !contractOperation.Deployer.Value.IsNullOrEmpty(),
+            "Invalid input deploying address.");
+        Assert(contractOperation.Salt != null && !contractOperation.Salt.Value.IsNullOrEmpty(), "Invalid input salt.");
+        Assert(contractOperation.CodeHash != null && !contractOperation.CodeHash.Value.IsNullOrEmpty(),
+            "Invalid input code hash.");
+        Assert(!contractOperation.Signature.IsNullOrEmpty(), "Invalid input signature.");
+
+        Assert(contractOperation.Version == version + 1, "Wrong input version.");
+        Assert(contractOperation.ChainId == Context.ChainId, "Wrong input chain id.");
+        Assert(contractOperation.CodeHash == codeHash, "Wrong input code hash.");
+
+        var hash = HashHelper.ComputeFrom(new ContractOperation
+        {
+            ChainId = contractOperation.ChainId,
+            CodeHash = contractOperation.CodeHash,
+            Deployer = contractOperation.Deployer,
+            Salt = contractOperation.Salt,
+            Version = contractOperation.Version
+        }.ToByteArray());
+        var publicKey = Context.RecoverPublicKey(contractOperation.Signature.ToByteArray(), hash.ToByteArray());
+
+        var recoveredAddress = Address.FromPublicKey(publicKey);
+
+        if (recoveredAddress == contractOperation.Deployer) return;
+
+        Assert(State.SignatoryMap[contractOperation.Deployer] == recoveredAddress,
+            "Wrong signature.");
+        RemoveSignatory(contractOperation.Deployer);
```

Comment:
This line doesn't belong to this method. This method is doing assertion. This line is updating the state. They don't belong together. It's violating single responsibility principle.